components/feature/download/src/main/java/mozilla/components/feature/download/DownloadManager.kt

```diff
@@ -0,0 +1,139 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.download
+
+import android.Manifest.permission.INTERNET
+import android.Manifest.permission.WRITE_EXTERNAL_STORAGE
+import android.content.Context
+import android.net.Uri
+import android.support.annotation.RequiresPermission
+import mozilla.components.browser.session.Download
+import mozilla.components.support.ktx.android.content.isPermissionGranted
+import mozilla.components.support.utils.DownloadUtils
+import android.content.BroadcastReceiver
+import android.content.Intent
+import android.content.IntentFilter
+import java.lang.IllegalArgumentException
+import android.app.DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED
+import android.app.DownloadManager.Request
+import android.app.DownloadManager.EXTRA_DOWNLOAD_ID
+import android.app.DownloadManager.ACTION_DOWNLOAD_COMPLETE
+import android.content.Context.DOWNLOAD_SERVICE
+
+typealias OnDownloadFinish = (Long) -> Unit
+typealias AndroidDownloadManager = android.app.DownloadManager
+
+/**
+ * Handles the interactions with the [AndroidDownloadManager].
+ * @property context A reference to [Context] applicationContext.
+ */
+class DownloadManager(private val context: Context) {
```

Comment:
nit: If we want this to be an applicationContext then let's make this a bit more obvious and name the parameter `applicationContext`.